### ISSUE TITLE
Optimizer rounds negative feeDiff values to zero

### ISSUE THREAD

**Issue by zobront** [stake-dao/only-boost#60]
In the optimal deposit formula in Optimizer.sol, we calculate the `feeDiff` between Convex and StakeDAO. `feeDiff` represents how much more profitable each dollar earned in Convex is than StakeDAO, taking into account fees and incentives.

Since StakeDAO's fee is 16% and Convex is 17%, and we are only taking into account Convex incentives, the possible range for `feeDiff` is `-1 <= feeDiff <= inf`.

When this value is calculated, we do the following:
```solidity
uint256 feeDiff = boost + stakeDaoTotalFee > convexTotalFee ? stakeDaoTotalFee + boost - convexTotalFee : 0;
```
In short, if `boost` (Convex incentives) is greater than the gap in fees, we perform the calculation. However, if `boost` is not sufficient to cover the gap in fees, we default to zero.

This creates a loss of precision in the formula for situations when Convex's incentive is less than 1%, which is due to happen in the coming years as their inflation rate slows down.

### Recommendation

To increase the precision, allow `feeDiff` to represent a negative number as low as `-1e16` and use that value in the computation.
